src/consensus/raft/raft.h

```diff
@@ -1184,58 +1206,80 @@ namespace raft
       }
 
       if (changed)
+      {
         create_and_remove_node_state();
+      }
     }
 
     void create_and_remove_node_state()
     {
       // Find all nodes present in any active configuration.
-      std::unordered_set<NodeId> active_nodes;
+      Configuration::Nodes active_nodes;
 
       for (auto& conf : configurations)
       {
-        for (auto node_id : conf.nodes)
-          active_nodes.insert(node_id);
+        for (auto node : conf.nodes)
+        {
+          active_nodes.emplace(node.first, node.second);
+        }
       }
 
-      // Find all nodes in the node state that are not present in any active
+      // Remove all nodes in the node state that are not present in any active
       // configuration.
       std::vector<NodeId> to_remove;
 
       for (auto& node : nodes)
       {
         if (active_nodes.find(node.first) == active_nodes.end())
+        {
           to_remove.push_back(node.first);
+        }
       }
 
       for (auto node_id : to_remove)
       {
+        if (state == Leader)
+        {
+          channels->destroy_channel(node_id);
+        }
         nodes.erase(node_id);
-        LOG_INFO_FMT("Removed node {}", node_id);
+        LOG_INFO_FMT("Removed raft node {}", node_id);
       }
 
-      for (auto node_id : active_nodes)
+      // Add all active nodes that are not already present in the node state.
+      bool self_is_active = false;
+
+      for (auto node_info : active_nodes)
       {
-        if (node_id == local_id)
+        if (node_info.first == local_id)
+        {
+          self_is_active = true;
           continue;
+        }
 
-        if (nodes.find(node_id) == nodes.end())
+        if (nodes.find(node_info.first) == nodes.end())
         {
           // A new node is sent only future entries initially. If it does not
           // have prior data, it will communicate that back to the leader.
           auto index = last_idx + 1;
-          nodes[node_id] = {0, index};
+          nodes.try_emplace(node_info.first, 0, index, node_info.second);
 
           if (state == Leader)
-            send_append_entries(node_id, index);
+          {
+            channels->create_channel(
```

Comment:
When a new `TRUSTED` node is added, only the leader should create a channel to the backups.